code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/interactions/UIEvent.kt

```diff
@@ -27,42 +28,44 @@ sealed interface UIEvent<T : AepUITemplate, S : AepCardUIState> {
      *
      * @param T represents UI template model associated like [SmallImageTemplate], which backs the composable on which the event has occurred.
      * @param S representing the state of the AEP card composable on which the event has occurred.
-     * @property _aepui The AEPUI associated with the display event.
+     * @property aepUi The [AepUI] associated with the display event.
      */
-    data class Display<T : AepUITemplate, S : AepCardUIState>(val _aepui: AepUI<T, S>) :
+    data class Display<T : AepUITemplate, S : AepCardUIState>(val aepUi: AepUI<T, S>) :
         UIEvent<T, S>
 
     /**
      * Event that represents a user interaction with a UI element.
      *
      * @param T represents UI template model associated like [SmallImageTemplate], which backs the composable on which the event has occurred.
      * @param S representing the state of the AEP card composable on which the event has occurred.
-     * @param action The type of interaction that occurred. This can be derived from the constants
-     * provided in [UIAction], such as [UIAction.CLICK] or [UIAction.EXPAND].
-     * @property aepUi The AEPUI associated with the interaction event, providing context about the UI component
-     * on which the interaction occurred.
+     * @property aepUi The [AepUI] associated with the interaction event, providing context about the UI component on which the interaction occurred.
+     * @property interactId The interaction identifier of the UI component on which the interaction occurred.
+     * @property action The [UIAction] that occurred.
+     * @property actionUrl The URL associated with the interaction event.
      *
      * The `Interact` event captures the different types of interactions that a user can have with a UI component,
      * like clicking a button or expanding a card. Limiting the interaction types ensures consistency in event
      * generation and handling.
      *
      * Example:
      * ```
-     * observer?.onEvent(AepUiEvent.Interact(ui, action = UIAction.CLICK))
+     * observer?.onEvent(AepUiEvent.Interact(ui, interactId = "purchaseID", action = UIAction.CLICK), actionUrl = "https://www.adobe.com")
      * ```
      */
     data class Interact<T : AepUITemplate, S : AepCardUIState>(
         val aepUi: AepUI<T, S>,
-        val action: String
+        val interactId: String?,
```

Comment:
nit: Can we rename this? This is the id of the field which was interacted with. Something like `interactedId`. Or maybe we can leave this out for now if its confusing

Reply:
this id is removed. the id will now be taken from the UIAction which occurred:
```kotlin
/**
 * Represents an action that can be performed on a UI component.
 */
sealed class UIAction {
    /**
     * UIAction to represent a click action that can be performed on a UI component
     * @property id unique identifier of the UI component
     * @property actionUrl optional URL to be opened when the UI component is clicked
     */
    data class Click(val id: String, val actionUrl: String?) : UIAction()
}
```